docs/tutorials/intro/executing_tasks.ipynb

```diff
@@ -0,0 +1,381 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Executing tasks\n",
+    "\n",
+    "Now that you know how to initialize brigade and work with the inventory let's see how we can leverage it to run tasks on groups of hosts.\n",
+    "\n",
+    "Brigade ships a bunch of tasks you can use directly without having to code them yourself. You can check them out [here](../../ref/tasks/index.rst).\n",
+    "\n",
+    "Let's start by executing the `ls -la /tmp` command on all the device in `cmh` of type `host`:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[1m\u001b[34m* host1.cmh ** changed : False *************************************************\u001b[0m\n",
+      "\u001b[0m\u001b[1m\u001b[36m---- remote_command ** changed : False  ----------------------------------------\u001b[0m\n",
+      "\u001b[0mtotal 8\n",
+      "drwxrwxrwt  2 root root 4096 Mar 11 19:17 .\n",
+      "drwxr-xr-x 24 root root 4096 Mar 11 17:28 ..\n",
+      "\u001b[0m\n",
+      "\u001b[0m\n",
+      "\u001b[0m\u001b[1m\u001b[34m* host2.cmh ** changed : False *************************************************\u001b[0m\n",
+      "\u001b[0m\u001b[1m\u001b[36m---- remote_command ** changed : False  ----------------------------------------\u001b[0m\n",
+      "\u001b[0mtotal 8\n",
+      "drwxrwxrwt  2 root root 4096 Mar 11 19:17 .\n",
+      "drwxr-xr-x 24 root root 4096 Mar 11 17:29 ..\n",
+      "\u001b[0m\n",
+      "\u001b[0m\n",
+      "\u001b[0m"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "AggregatedResult: print_result"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from brigade.plugins.tasks import commands, text\n",
+    "\n",
+    "cmh_hosts = brg.filter(site=\"cmh\", role=\"host\")\n",
+    "\n",
+    "result = cmh_hosts.run(commands.remote_command,\n",
+    "                       command=\"ls -la /tmp\")\n",
+    "\n",
+    "cmh_hosts.run(text.print_result,\n",
+    "              num_workers=1,\n",
+    "              data=result,\n",
+    "              vars=[\"stdout\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So what have we done here? First we have imported the `commands` and `text` modules. Then we have narrowed down brigade to the hosts we want to operate on. Once we have selected the devices we wanted to operate on we have run two tasks:\n",
+    "\n",
+    "1. The task `commands.remote_command` which runs the specified `command` in the remote device.\n",
+    "2. The task `text.print_result`  which just prints on screen the result of an executed task (note this task needs to be serialized to avoid mangling the output so `num_workers=1` is specified).\n",
+    "\n",
```

Comment:
Add a section describing where "stdout" comes from and how data is passed around using the result object?